lib/fetchers/helpers.ts

```diff
@@ -0,0 +1,52 @@
+import { FetchParams, RequestOptions } from "./types"
+
+/**
+ * Build URLSearchParams instance with support for arrays of values.
+ *
+ * By default, URLSearchParams will join arrays of values with a comma. This is
+ * not desirable for our use case. Instead, we want arrays of values to be
+ * represented as multiple entries with the same key.
+ */
+export const buildParams = (data: FetchParams) => {
+  let params: URLSearchParams
+
+  if (typeof data === "string" || data === undefined) {
+    params = new URLSearchParams(data)
+  } else {
+    params = new URLSearchParams()
+
+    Object.entries(data).forEach(([key, value]) => {
+      if (Array.isArray(value)) {
+        value.forEach(inner => {
+          params.append(key, inner.toString())
+        })
+      } else {
+        params.append(key, value.toString())
+      }
+    })
+  }
+
+  return params
+}
```

Comment:
Is this something that maybe belongs to a general way of handling url parameters?
We do something similar in the search filters right.
Consider consolidating the code if possible.

Reply:
I feel like this is quite specifically targeted to creating search parameters for fetchers. Filter params we work with in a different way where we need to toggle them on and off. 
You're right that this function is a more generic one, but I don't think we can merge it / use it with filters.